src/main/java/org/example/springbootintro/dto/book/validator/BookSearchParametersValidator.java

```diff
@@ -0,0 +1,65 @@
+package org.example.springbootintro.dto.book.validator;
+
+import org.example.springbootintro.dto.book.BookSearchParametersDto;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+@Component
+public class BookSearchParametersValidator implements Validator {
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return BookSearchParametersDto.class.equals(clazz);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
```

Comment:
I want to use or only title or author or both. If I will use annotations @NotEmpty or something else, it will fail. Bohdan said that I need to create custom.